### ISSUE TITLE
ERC721 function _transfer, only allows owner

### ISSUE THREAD

**Issue by leon-do** [OpenZeppelin/openzeppelin-contracts#2871]
I'm trying to use the `approve` method to allow addressB to send on my behalf. This is works, but when I switch to addressB and `safeTransferFrom`, I get `ERC721: transfer of token that is not own`

`safeTransferFrom` checks `_isApprovedOrOwner` first which makes sense. Either owner or approved can transfer. https://github.com/OpenZeppelin/openzeppelin-contracts/blob/01f2ff1ba1220d7d3a0642014d30543bfcbb41ee/contracts/token/ERC721/ERC721.sol#L183

But when it calls `_transfer`, the require is more strict. It only allows the owner, hence the bug. https://github.com/OpenZeppelin/openzeppelin-contracts/blob/01f2ff1ba1220d7d3a0642014d30543bfcbb41ee/contracts/token/ERC721/ERC721.sol#L334

Is this intended?

**💻 Environment**

**📝 Details**

**🔢 Code to reproduce bug**

**Comment by Amxx** [comment]
Yes, this is part of [the standard requirements](https://eips.ethereum.org/EIPS/eip-721):

```
    /// @notice Transfer ownership of an NFT -- THE CALLER IS RESPONSIBLE
    ///  TO CONFIRM THAT `_to` IS CAPABLE OF RECEIVING NFTS OR ELSE
    ///  THEY MAY BE PERMANENTLY LOST
    /// @dev Throws unless `msg.sender` is the current owner, an authorized
    ///  operator, or the approved address for this NFT. Throws if `_from` is
    ///  not the current owner. Throws if `_to` is the zero address. Throws if
    ///  `_tokenId` is not a valid NFT.
    /// @param _from The current owner of the NFT
    /// @param _to The new owner
    /// @param _tokenId The NFT to transfer
    function transferFrom(address _from, address _to, uint256 _tokenId) external payable;
```
